contributed_definitions/nyaml/NXapm.yaml

```diff
@@ -559,2745 +392,557 @@ NXapm(NXobject):
-        temperature(NX_FLOAT):
-          exists: optional
-          unit: NX_TEMPERATURE
-          doc: |
-            The best estimate, at experiment time, for the temperature at the
-            sample base (furthest point along sample apex and holding assembly
-            that is removable from the sample stage).
-          dimensions:
-            rank: 1
-            dim: [[1, n_ions]]
-        (NXcsg):
-          exists: optional
-      
-      # evaporation control in which context is it used?
-      # NEW ISSUE: begin add and support I/O of further details
-      # NEW ISSUE: with Shyam Katnagallu fix that so far the application definition does not really cover fim as there is no place to store values for a gas injection system and a (partial) pressure sensor for the imaging gas it should be clarified in the docstring of the pressure field if this measured either a chamber total of a species partial pressure
-      # NEW ISSUE: add NXapm_energy_analyzer, a voltage grid like done in Rouen/GPM
-      analysis_chamber(NXchamber):
-        name:
-          exists: optional
-        (NXfabrication):
-          exists: optional
-          vendor:
-            exists: recommended
-          model:
-            exists: recommended
-          identifier:
-            exists: recommended
-          capabilities:
-            exists: optional
-        description:
-          exists: optional
-        pressure(NX_FLOAT):
-          unit: NX_PRESSURE
-          doc: |
-            Average pressure in the analysis chamber.
-        (NXcsg):
-          exists: optional
-      buffer_chamber(NXchamber):
-        exists: optional
-        name:
-          exists: optional
-        (NXfabrication):
-          exists: optional
-          vendor:
-            exists: recommended
-          model:
-            exists: recommended
-          identifier:
-            exists: recommended
-          capabilities:
-            exists: optional
-        description:
-          exists: optional
-        pressure(NX_FLOAT):
-          unit: NX_PRESSURE
-          doc: |
-            Average pressure in the buffer chamber.
-        (NXcsg):
-          exists: optional
-      load_lock_chamber(NXchamber):
-        exists: optional
-        name:
-          exists: optional
-        (NXfabrication):
-          exists: optional
-          vendor:
-            exists: recommended
-          model:
-            exists: recommended
-          identifier:
-            exists: recommended
-          capabilities:
-            exists: optional
-        description:
-          exists: optional
-        pressure(NX_FLOAT):
-          unit: NX_PRESSURE
-          doc: |
-            Average pressure in the load_lock_chamber.
-        (NXcsg):
-          exists: optional
-      getter_pump(NXpump):
-        exists: optional
-        design:
+              vendor(NX_CHAR):
+              model(NX_CHAR):
+            wavelength(NX_FLOAT):
```

Comment:
`wavelength` does not exists in the `NXsource` base class. Note that in MPES, we have gone away from storing beam properties like `wavelength` on NXsource and rather create `NXbeam` instances. Might be good to align with that approach also here.

Reply:
Agree to some extent, in the long run yes, wavelength should be made a property of an instance NXbeam using the path modelling approach from Ron.
For atom probe right now in virtually all cases, unless you are the single person at Cameca who does the laser system engineering you do not have access to such details. 
This is why I kept for now wavelength in source but
i) made a comment about moving at some point to NXbeam
ii) and indeed made a doc string because in Nxsource there wont be any longer a wavelength so it cannot just be composed but has to be defined as a new concept right now

However keep in mind that NXbeam still on fairmat has the following doc string:
"Properties of the neutron or X-ray beam at a given location." So reviewers have an argument that we use a concept that is neutron or Xray for electron ... which should not be the case, right?

Reply:
In #157, we still have 
```
NXsource:
  [...]
  wavelength(NX_FLOAT):
    unit: NX_WAVELENGTH
    doc: |
      The wavelength of the radiation emitted by the source.
```
Shall we keep it there? In that case, you could drop the docstring here. If not, I will remove it from NXsource in #157. By the way, could you have a look there to see that you are not using any of the other fields that that PR would remove from NXsource?

I suggest that we propose to change the docstring of NXbeam then. If you agree, I will make a new PR for this.

Reply:
Yeah do so. Thank you.